### PR TITLE
Enabled multiple usage of AutoMapAttribute

### LINKED ISSUE #3049: Allow multiple uses of AutoMapAttribute
We frequently define maps from multiple sources to a single destination. It'd be great, and clean up our code substantially, to be able to do so using the new attribute mapping, but the `AutoMapAttribute` class doesn't allow multiple usage as the relevant flag in `AttributeUsage` is not set.

As an example, we may wish to instantiate a Command class from either an API call model, or a UI View Model:

```csharp
public class Command()
{
    // members

    public class AutoMapperProfile : Profile
    {
        public AutoMapperProfile()
        {
            CreateMap<ApiModel, Command>();
            CreateMap<ViewModel, Command>();
        }
    }
}
```

Ideally, we'd like to be able to do

```csharp
[AutoMap(typeof(ApiModel))]
[AutoMap(typeof(ViewModel))]
public class Command()
{
    // members
}
```

I've had a very quick look, and it appears that setting the appropriate flag on the `AutoMapAttribute.cs` class and changing how those attributes are loaded in `MapperConfigurationExpression.cs` is what's required to allow this.

### PR DESCRIPTION
**Pull request by Sharkwald**
Fixes #3049.
This is a rough first pass to enable this. All existing tests are green, but it needs some specifically covering the multiple attribute use scenario.

### CODE REVIEW

**Comment by Sharkwald** [comment]
Added a simple test to describe the scenario I have in mind. Are any more required?

File: src/AutoMapper/Configuration/MapperConfigurationExpression.cs [hunk 1]
```diff
@@ -106,17 +106,23 @@ private void AddMapsCore(IEnumerable<Assembly> assembliesToScan)
...
-                var autoMapAttribute = type.GetCustomAttribute<AutoMapAttribute>();
-                if (autoMapAttribute != null)
+                var autoMapAttributes = type.GetCustomAttributes<AutoMapAttribute>();
+                foreach (var autoMapAttribute in autoMapAttributes)
                 {
-                    var mappingExpression = (MappingExpression) autoMapAttributeProfile.CreateMap(autoMapAttribute.SourceType, type);
-                    autoMapAttribute.ApplyConfiguration(mappingExpression);
-
-                    foreach (var memberInfo in type.GetMembers(BindingFlags.Public | BindingFlags.Instance))
+                    if (autoMapAttribute != null)
```
Comment (lbargaoanu) [inline src/AutoMapper/Configuration/MapperConfigurationExpression.cs]: I think the null check can be removed.

Comment (Sharkwald) [reply src/AutoMapper/Configuration/MapperConfigurationExpression.cs]: So it can.

**Comment by Sharkwald** [comment]
The test failures are being caused by the change to the test base class; because the name is not being specified (as `nameof(AutoMapAttribute)`) it seems to be validating globally and tripping on some failure conditions set elsewhere in the test code.

**Comment by lbargaoanu** [comment]
I'll take a look.

**Comment by lbargaoanu** [comment]
I just reverted :) I guess those scanning tests should live in their own assembly, because now it just scans everything.

**Comment by Sharkwald** [comment]
Brilliant :D.

**Comment by lbargaoanu** [comment]
We need your signature :)

**Comment by Sharkwald** [comment]
Apologies, that should be it now!

**Pull request merged**